src/Tasks/ComReference.cs

```diff
@@ -407,21 +407,32 @@ internal static string StripTypeLibNumberFromPath(string typeLibPath, FileExists
         private static string GetModuleFileName(IntPtr handle)
         {
             bool success = false;
-            var buffer = new StringBuilder();
+            char[] buffer = null;
+            string output = string.Empty;
 
             // Try increased buffer sizes if on longpath-enabled Windows
             for (int bufferSize = NativeMethodsShared.MAX_PATH; !success && bufferSize <= NativeMethodsShared.MaxPath; bufferSize *= 2)
             {
-                buffer.EnsureCapacity(bufferSize);
-
-                var handleRef = new System.Runtime.InteropServices.HandleRef(buffer, handle);
-                int pathLength = NativeMethodsShared.GetModuleFileName(handleRef, buffer, buffer.Capacity);
+                buffer = System.Buffers.ArrayPool<char>.Shared.Rent(bufferSize);
+                try
+                {
+                    var handleRef = new System.Runtime.InteropServices.HandleRef(buffer, handle);
+                    int pathLength = NativeMethodsShared.GetModuleFileName(handleRef, buffer, bufferSize);
 
-                bool isBufferTooSmall = ((uint)Marshal.GetLastWin32Error() == NativeMethodsShared.ERROR_INSUFFICIENT_BUFFER);
-                success = pathLength != 0 && !isBufferTooSmall;
+                    bool isBufferTooSmall = (uint)Marshal.GetLastWin32Error() == NativeMethodsShared.ERROR_INSUFFICIENT_BUFFER;
+                    success = pathLength != 0 && !isBufferTooSmall;
+                    if (success)
+                    {
+                        output = new string(buffer, 0, pathLength);
```

Comment:
super-nit:

```suggestion
                        return new string(buffer, 0, pathLength);
```

and `return string.Empty;` at the end of the method.